plug-ins/crossfadetracks.ny

```diff
@@ -80,6 +78,7 @@ audio clip, fade in, otherwise fade out."
       (setf out-dist (min out-dist (abs (- end (second (nth i clips)))))))
     (if (< in-dist out-dist) 'in 'out)))
 
+
 (if (< (length (get '*selection* 'tracks)) 2)
     (format nil (_ "Error.~%Select 2 (or more) tracks to crossfade."))
-    (crossfade type direction curve))
+    (crossfade))
```

Comment:
Why can these arguments now be omitted?

Reply:
> Why can these arguments now be omitted?

Because the function `crossfade` does not need them.
`TYPE`, `DIRECTION` and `CURVE` are all global variables. They have to be global because they are defined by the controls:
```
$control TYPE (_ "Fade TYPE") choice (
    ("ConstantGain" (_ "Constant Gain"))
    ("ConstantPower1" (_ "Constant Power 1"))
    ("ConstantPower2" (_ "Constant Power 2"))
    ("CustomCurve" (_ "Custom Curve"))) 0
$control CURVE (_ "Custom curve") real "" 0 0 1
$control DIRECTION (_ "Fade direction") choice (
    (_ "Automatic")
    ("OutIn" (_ "Alternating Out / In"))
    ("InOut" (_ "Alternating In / Out"))) 0
```

The point of this pull request is that since we **must** use global variables in Nyquist plug-ins that have controls, then it is safer and more readable to implement a convention where we treat them as global constants, and mark them as such by using upper case.